composer/utils/dist.py

```diff
@@ -454,6 +455,23 @@ def get_sampler(dataset: torch.utils.data.Dataset, *, drop_last: bool = False, s
     )
 
 
+@contextmanager
+def local_rank_zero_download_and_wait(expected_file_path: str):
+    """Context manager to wait for a file to exist on all ranks except local rank zero.
+
+    It is expected that the file will be created by local rank zero.
+
+    Args:
+        expected_file_path (str): The file to wait for existence of
+    """
+    local_rank = get_local_rank()
+    if local_rank != 0:
+        while not os.path.exists(expected_file_path):
+            time.sleep(0.1)
+
+    yield
```

Comment:
Can we instead use `run_local_rank_zero_first`? It's the same functionality -- it will run the context manager for rank 0 and then only continue on other ranks until after the context manager ends aka download is finished and data exists locally.

Reply:
We can't use that because it requires dist to be initialized, which we don't want to force a user to do before creating their dataset.

Reply:
Can you specify in docstring the difference and when to use this one over the other one?

Reply:
Yup, will do